meson.build

```diff
@@ -341,6 +341,7 @@ endif
 basic_disabled_warnings = [
         '-Wno-missing-field-initializers',
         '-Wno-unused-parameter',
+        '-Wno-nonnull-compare',
```

Comment:
hmm, at which of our calls does that actually trigger?

Reply:
This check complains everywhere where we use `int x[static X]` in function declaration and then call `assert()` on it:

```
../src/basic/fd-util.h: In function ‘format_proc_fd_path’:
../src/fundamental/macro-fundamental.h:71:41: warning: ‘nonnull’ argument ‘buf’ compared to NULL [-Wnonnull-compare]
   71 | #define _unlikely_(x) (__builtin_expect(!!(x), 0))
      |                                         ^~~~~
../src/basic/macro.h:148:21: note: in expansion of macro ‘_unlikely_’
  148 |                 if (_unlikely_(!(expr)))                                \
      |                     ^~~~~~~~~~
../src/basic/macro.h:166:22: note: in expansion of macro ‘assert_message_se’
  166 | #define assert(expr) assert_message_se(expr, #expr)
      |                      ^~~~~~~~~~~~~~~~~
../src/basic/fd-util.h:129:9: note: in expansion of macro ‘assert’
  129 |         assert(buf);
      |         ^~~~~~
```

https://github.com/systemd/systemd/blob/02b346d64f3667ebb857f064a3e8ad84682d0e5f/src/basic/fd-util.h#L128-L133

It's actually quite a lot of warnings, so I'm not completely sure if we want drop the asserts or disable the warning just for the assert stuff via using the _Pragma() diagnostic push stuff.